lib/br_danfe/danfe_lib/infadic.rb

```diff
@@ -6,131 +6,99 @@ class Infadic
+      def render_volumes
+        InfadicVol.new(@xml, @pdf).render
       end
 
-      def numerify(value)
-        Helper.numerify(value) if value != ""
+      def render_additional_data(y_position)
+        data = generate_additional_data
+        @pdf.ibox 2.65, 12.45, 0.75, y_position, '', data, additional_data_style
       end
 
-      def render_extra_volume(det, y)
-        render_field "qVol", det, 0.70, 0.75, 0.70, 1.04, y, :text
-        render_field "esp", det, 0.50, 1.35, 3.00, 1.75, y, :text
-        render_field "marca", det, 0.70, 4.15, 2.00, 4.75, y, :text
-        render_field "nVol", det, 1.00, 6.10, 1.00, 6.70, y, :text
-        render_field "pesoB", det, 1.30, 7.00, 1.30, 7.00, y, :numeric
-        render_field "pesoL", det, 0.90, 8.50, 1.50, 8.50, y, :numeric
+      def generate_additional_data
+        additional_data = []
+        additional_data.push(complementary_content) if complementary?
+        additional_data.push(address_content) if address?
+        additional_data.push(difal_content) if difal?
+        additional_data.join(' * ')
       end
 
-      def render_field(field, det, w1, x1, w2, x2, y, kind)
-        label = I18n.t("danfe.infAdic.vol.#{field}")
-        value = det.css(field).text
-
-        @pdf.ibox 0.35, w1, x1, y, "", label, style_normal
+      def additional_data_style
+        { size: 6, valign: :top, border: 0 }
+      end
 
-        if kind == :numeric
-          @pdf.inumeric 0.35, w2, x2, y, "", value, style_decimal
-        else
-          @pdf.ibox 0.35, w2, x2, y, "", value, style_italic
+      def y_position(volumes_number)
+        if volumes_number > 1
+          return Y_POSITION + 0.30 + volumes_number * 0.15 + 0.2
         end
+        Y_POSITION + 0.30
       end
 
-      def style_normal
-        { size: 4, border: 0 }
+      def additional_data?
+        complementary? || address? || difal?
       end
 
-      def style_italic
-        style_normal.merge({ style: :italic })
+      def complementary_content
+        @xml['infAdic/infCpl'].to_s
       end
 
-      def style_decimal
-        style_italic.merge({ decimals: 3 })
+      def complementary?
+        @xml['infAdic/infCpl'].to_s != ''
       end
 
-      def render_info_cpl_with_others(y)
-        @pdf.ibox 1.65, 12.45, 0.75, y + 0.30, "", I18n.t("danfe.infAdic.others"), { size: 6, valign: :top, border: 0 }
-        @pdf.ibox 1.65, 12.45, 0.75, y + 0.50, "", @xml["infAdic/infCpl"], { size: 5, valign: :top, border: 0 }
+      def address_content
+        "Endereço: #{Helper.generate_address @xml}"
       end
 
-      def render_difal
-        @pdf.ibox 2.65, 12.45, 0.75, @y_position, I18n.t("danfe.infAdic.infCpl"), "", { size: 8, valign: :top }
-
-        y = Y_POSITION + 0.20
-        @pdf.ibox 1.65, 12.45, 0.75, y, "", difal_content, { size: 5, valign: :top, border: 0 }
-
-        y += 0.10
+      def address?
+        Helper.address_is_too_big @pdf, Helper.generate_address(@xml)
```

Comment:
Só vai ser impresso o endereço se ele tiver mais que 319 caracteres?

Reply:
Não modifiquei essa lógica, esta igual anteriormente, a ideia é só aparecer o endereço em informações adicionais se ele for muito grande para aparecer no topo do Danfe.